pyproject.toml

```diff
@@ -37,14 +37,14 @@ optional = true
 optional = true
 
 [tool.poetry.group.development.dependencies]
-pyspark = ">2"
+pyspark = "^3.5.0"
```

Comment:
I suggest incrementing a major version of quinn in a new release. Dropping spark 2 is a big change. Also, a CI/CD should be allined